src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
QPC ticks always increment by 1 (*). That's why you divide by the frequency - the frequency tells you the number of ticks per second. So "QPC uses 32 bits for one second" and "1 microsecond resolution" are contradictory. If the resolution is 1 microsecond then QPC will increment by 1e6 every second, i.e. it will use 17 bits for one second.

Also, a resolution of 1 microsecond is the *worst* that QPC can give. The frequency is commonly about 1/1000th of the processor speed, so a 3GHz processor will have about 3MHz frequency = 0.3 microseconds resolution. On Hyper-V guests, the frequency is fixed at 10MHz = 0.1 microsecond resolution.

Take my machine as an example. It has a QPF of 3.3E6 so a 300ns resolution. 3.3E6 is indeed a 22-bit number, which matches up with how I said the value of QPC differs by 22 bits after one second.

(*) This is not true of variant TSCs, motherboard timers, etc. But either way the implication that resolution is independent of the frequency is not correct.